core/web/resolver/mutation.go

```diff
@@ -393,3 +393,96 @@ func (r *Resolver) DeleteBridge(ctx context.Context, args struct {
 
 	return NewDeleteBridgePayload(&bt, nil), nil
 }
+
+func (r *Resolver) ApproveJobProposal(ctx context.Context, args struct {
+	ID graphql.ID
+}) (*ApproveJobProposalPayloadResolver, error) {
+	rsv, err := r.executeJobProposalAction(ctx, jobProposalAction{
+		args.ID, approve,
+	})
+
+	prv, ok := rsv.(*ApproveJobProposalPayloadResolver)
```

Comment:
I'd prefer duplication over type conversions and passing resolvers down. Errors which occur here would only surface at runtime, instead of compile time

Reply:
Understood, I changed the implementation so it doesn't use conversions.